modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -162,7 +163,13 @@ void onScrollResponse(SearchResponse searchResponse) {
         }
         setScroll(searchResponse.getScrollId());
         if (searchResponse.getShardFailures() != null && searchResponse.getShardFailures().length > 0) {
-            startNormalTermination(emptyList(), unmodifiableList(Arrays.asList(searchResponse.getShardFailures())));
+            startNormalTermination(emptyList(), unmodifiableList(Arrays.asList(searchResponse.getShardFailures())),
+                    searchResponse.isTimedOut());
+            return;
+        }
+        // Timeouts aren't shard failures but we still need to bass them back to the user.
+        if (searchResponse.isTimedOut()) {
```

Comment:
This could be collapsed into the previous `if` statement:

```java
if ((searchResponse.getShardFailures() != null && searchResponse.getShardFailures().length > 0) || searchResponse.isTimedOut()) {
    ....
}